### PR TITLE
feat: add an option to set default headers for all queries/updates

### PR DESCRIPTION
**Pull request by tpluscode**
This way it will be easier to set `Authorization` header once for example

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -370,6 +370,40 @@ describe('sparql-http-client', function () {
...
+      nock('http://example.org')
+          .get('/sparql?query=' + encodeURIComponent(simpleConstructQuery))
+          .reply(200, function () {
+            assert.equal(this.req.headers.authorization, 'Bearer foo')
+          })
+
+      return endpoint.constructQuery(simpleConstructQuery)
+    })
+
+    it('should override default headers with passed to function', function () {
```
Comment (vhf) [inline test/test.js]:
```suggestion
    it('should override default headers with argument passed to function', function () {
```
I had a hard time understanding it, I suggest this change

**Comment by tpluscode** [comment]
@vhf 70f277b

> I think most editors have auto detection for the given settings, WebStorm even detects standard.

You sure? My webstorm stubbornly want to indent stuff with 4 spaces...

**Comment by vhf** [comment]
> @vhf [70f277b](https://github.com/zazuko/sparql-http-client/commit/70f277b21d31526d4e7cada88900898685b85b9d)

Nice, thanks!

> > I think most editors have auto detection for the given settings, WebStorm even detects standard.
> 
> You sure? My webstorm stubbornly wants to indent stuff with 4 spaces...

I don't know about IDEs but my editor(s) don't detect things, that's why I use .editorconfig in a code sharing context, except when there is a linter in place that already checks (and fixes) things such as trailing spaces, \n at the end of the file, indentation, etc.

**Comment by tpluscode** [comment]
You're not gonna like this @bergos but I installed eslint instead of standard so that I can reduce the severity of deprecation rule from error to warning. Not ideal, but it's not the time and place to fix all of those test IMO

**Comment by tpluscode** [comment]
> except when there is a linter in place

Even with a linter, an editor using editor config might proactively keep the right formatting when doing code-completion. I certainly don't welcome constant need to reformat code because the IDE generate it poorly formatted and the immediately highlights linting errors

**Comment by vhf** [comment]
Agreed. What I was trying to convey is that Standard isn't strict enough to my taste, for instance I like to enforce a newline at EOF and other things related to maintaining a consistent codebase without having to think about it (and a bit unrelated to strictly checking the code).

**Comment by bergos** [comment]
If you remove the stuff about editor settings and linting, I will take care of adding `standard` in a separate PR.

**Comment by tpluscode** [comment]
I'm not stopping you. Please set up standard on master and I will rebase and clean up tis branch

**Pull request merged**